src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -112,19 +112,29 @@ public ProgrammingExerciseService(ProgrammingExerciseRepository programmingExerc
     }
 
     /**
-     * Notifies all particpations of the given programmingExercise about changes of the test cases.
+     * Notifies all participations of the given programmingExercise (including the template & solution participation!) about changes of the test cases.
+     * This method creates submissions for the participations so that the result when it comes in can be mapped to them.
      *
-     * @param programmingExercise The programmingExercise where the test cases got changed
+     * @param exerciseId of programming exercise the test cases got changed.
      */
-    public void notifyChangedTestCases(ProgrammingExercise programmingExercise, Object requestBody) {
-        for (StudentParticipation participation : programmingExercise.getParticipations()) {
-
-            ProgrammingExerciseStudentParticipation programmingExerciseStudentParticipation = (ProgrammingExerciseStudentParticipation) participation;
+    @Transactional
+    public void notifyChangedTestCases(Long exerciseId, Object requestBody) throws EntityNotFoundException {
+        Optional<ProgrammingExercise> exerciseOpt = programmingExerciseRepository.findById(exerciseId);
+        if (!exerciseOpt.isPresent())
+            throw new EntityNotFoundException("Programming exercise with id " + exerciseId + " not found.");
+        ProgrammingExercise programmingExercise = exerciseOpt.get();
+        // All student repository builds and the builds of the template & solution repository must be triggered now!
```

Comment:
I think this was a regression from the remodel PR: We didn't create submissions for the template/solution participations here anymore.